### DISCUSSION TITLE
dashboard for climate data while awake

### DISCUSSION THREAD

**Discussion by jfliu1** [teslamate-org/teslamate#3316] [Ideas]
Does teslamate pull climate data while the car is awake (i.e., no need to wake up the car while sleeping)? I'm particularly curious about the cabin temperature reading, e.g., when the car is parked and cabin heat protection is on. It'd be great to have a dashboard showing the outside and cabin temperature data.

thanks!

**Comment by cwanja** [comment]
Through the use of the Trip dashboard, I could construct a panel that would show the climate status during the time range of the dashboard. Which can either be:

- Last three drives when accessing the Trip dashboard natively
- A time range when another dashboard is sending data to the Trip dashboard (e.g. from Timeline - click on "Parking" status and that sends you to the Trip dashboard with the timeframe that the car was parked and thus you can see the climate of the car.

![Screenshot 2023-08-16 at 12 17 42 PM](https://github.com/adriankumpf/teslamate/assets/54645561/beedd5b5-cfe7-4540-9868-a2d1f005c49b)

![Screenshot 2023-08-16 at 12 22 45 PM](https://github.com/adriankumpf/teslamate/assets/54645561/c4bd4174-3e34-4a40-8da2-1ab79a680b47)

![Screenshot 2023-08-16 at 12 29 20 PM](https://github.com/adriankumpf/teslamate/assets/54645561/e180dca9-4385-4a34-bc67-5b391a1c716a)

![Screenshot 2023-08-16 at 12 32 26 PM](https://github.com/adriankumpf/teslamate/assets/54645561/556f0594-805f-4e79-b15a-dbc2dea746da)

From a rough idea to implemented (fourth screenshot). In the fourth screenshot you can see the inside and outside temperature, if the climate is on or off and the fan status. The challenge is - it does not look like TeslaMate today captures if **cabin overheat is on or off** and that is separate from the climate being on or off.

To answer you question - you could show inside and outside temperatures for a time range and when the actual climate controls are on or off. But currently you are unable to capture when cabin overheat protection is on during a time range.

**Comment by jfliu1** [comment]
Thanks, that's helpful!

I assume when the cabin overheat protection is on (e.g., when fan and/or AC operates to cool the cabin), the car will become awake and the temperature reading will be pulled by teslamate. Is that correct?

**Reply by cwanja** [reply]
That is correct. The car cannot sleep and actively be cooling at the same time. Thus TeslaMate would be capturing the temperatures. I am constructing a PR for adding the cabin overheat to the database. Couple of points:
1. It might never get added. But trying. 
2. If it is added, any historical data will be null and useless. It would only capture the data from when the PR is implemented and a user updates TeslaMate.

**Comment by cwanja** [comment]
#3317 see what happens 🤷🏼
